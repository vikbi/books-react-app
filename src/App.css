.book-app {
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 500px;
  margin: 0 auto;
  border: 1px solid;
}

.header {
    display: flex;
    justify-content: flex-start;
    align-items: flex-start;
    height: 45px;
    padding: 5px;
    border: 1px solid;
    background-color: #fff8dc;
    font-size: 25px;
    font-weight: 600;
    align-items: center;
}

.sub-header {
  padding-left: 5px;
  height: 40px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  border: 1px solid;
}

.book-detail {
  display: flex;
  flex-direction: row;
  max-height: 100px;
  border: 1px dashed rgb(152, 149, 149);
  padding: 2px;
}

.book-logo {
  height: 75px;
  width: 50px;
}

.book-header {
  flex-direction: column;
  display: flex;
  margin-left: 5px;
}
.book-list {
  height: 500px;
  overflow-y: scroll;
}
.button-clear {
    padding: 5px;
    margin-right: 5px;
    font-weight: 600;
    background-color: #f7cf93;
    border-radius: 4px;
    cursor: pointer;
}
.loader {
  background-color: black;
  color: white;
  height: 40px;
  align-items: center;
  display: flex;
  justify-content: center;
  position: absolute;
  z-index: 1000;
  bottom: 11%;
  left: 45%;
  width: 80px;
  border-radius: 5px;
}