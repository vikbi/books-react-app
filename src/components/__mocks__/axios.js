export const mockResponse = {
    data: {
      results: [
        {
            id: '1',
            title: 'test-title',
            formats: {'image/jpeg' : 'image-link'},
            authors: [{name: 'author1'}],
            is_checked: false
        },
        {
            id: '2',
            title: 'test-title2',
            formats: {'image/jpeg' : 'image-link2'},
            authors: [{name: 'author2'}],
            is_checked: false
        },
        {
            id: '3',
            title: 'test-title',
            formats: {'image/jpeg' : 'image-link'},
            authors: [{name: 'author3'}],
            is_checked: false
        }
      ]
    }
}
export default {
    get: jest.fn().mockResolvedValue(mockResponse)
}