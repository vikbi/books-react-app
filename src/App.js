import './App.css';
import BookList from './components/BookList';

function App() {
  return (
    <div className='book-app'>    
      <BookList />
    </div>
  );
}

export default App;
