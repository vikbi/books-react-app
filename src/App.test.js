import { cleanup, render, screen } from '@testing-library/react';
import renderer from "react-test-renderer";
import App from './App';
import axios, { mockResponse } from './components/__mocks__/axios';

describe('App', () => {

  afterEach(cleanup)

  beforeEach(async() => await axios.get.mockResolvedValue(mockResponse));

  it('renders books app', () => {
    render(<App />);
    const linkElement = screen.getByText(/books/i);
    expect(linkElement).toBeInTheDocument();
  });
  
  it("Matches DOM Snapshot", () => {
    const domTree = renderer.create(<App />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
})
