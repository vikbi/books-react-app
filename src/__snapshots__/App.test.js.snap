// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Matches DOM Snapshot 1`] = `
<div
  className="book-app"
>
  <div
    className="header"
  >
    Books
  </div>
  <div
    className="sub-header"
  >
    <span>
      0
       books 
      <strong>
        (
        0
         selected)
      </strong>
    </span>
    <button
      className="button-clear"
      onClick={[Function]}
    >
      clear selection
    </button>
  </div>
  <div
    className="book-list"
    onScroll={[Function]}
  >
    <div />
  </div>
</div>
`;

exports[`Matches DOM Snapshot 1`] = `
<div
  className="book-app"
>
  <div
    className="header"
  >
    Books
  </div>
  <div
    className="sub-header"
  >
    <span>
      0
       books 
      <strong>
        (
        0
         selected)
      </strong>
    </span>
    <button
      className="button-clear"
      onClick={[Function]}
    >
      clear selection
    </button>
  </div>
  <div
    className="book-list"
    onScroll={[Function]}
  >
    <div />
  </div>
</div>
`;
